src/registry/types.ts

```diff
@@ -25,6 +25,7 @@ export interface MetadataRegistry {
  * Metadata type definition in the registry.
  */
 export interface MetadataType {
+  requiresParent?: boolean;
```

Comment:
add the jsdoc for that the option does, like all the others (if we decide to use the new option).  Otherwise, modify isAddressable's explanation